pytube/contrib/channel.py

```diff
@@ -183,19 +230,38 @@ def _extract_videos(raw_json: str) -> Tuple[List[str], Optional[str]]:
             # if there is an error, no continuation is available
             continuation = None
 
+        # only extract the video ids from the video data
+        videos_url = []
+        try:
+            # Extract id from videos
+            for x in videos:
+                videos_url.append(f"/watch?v="
+                                  f"{x['richItemRenderer']['content']['videoRenderer']['videoId']}")
+        except (KeyError, IndexError, TypeError):
+            # Extract id from short videos
+            for x in videos:
+                videos_url.append(f"/watch?v="
+                                  f"{x['richItemRenderer']['content']['reelItemRenderer']['videoId']}")
+
         # remove duplicates
-        return (
-            uniqueify(
-                list(
-                    # only extract the video ids from the video data
-                    map(
-                        lambda x: (
-                            f"/watch?v="
-                            f"{x['gridVideoRenderer']['videoId']}"
-                        ),
-                        videos
-                    )
-                ),
-            ),
-            continuation,
-        )
+        return uniqueify(videos_url), continuation
+
+    @property
+    def videos(self) -> Iterable[YouTube]:
+        """Yields YouTube objects of videos in this channel
+
+        :rtype: List[YouTube]
+        :returns: List of YouTube
+        """
+        self._html_page = self.videos_url  # Set video tab
+        return DeferredGeneratorList(self.videos_generator())
+
+    @property
+    def shorts(self) -> Iterable[YouTube]:
+        """Yields YouTube objects of short videos in this channel
+
+       :rtype: List[YouTube]
+       :returns: List of YouTube
+       """
+        self._html_page = self.shorts_url  # Set shorts tab
```

Comment:
> `self._html_page = self.shorts_url  # Set shorts tab`

Unfortunately, it won't work correctly as multiple properties(`video_urls`, `last_updated`, `title`) are cached and won't be updated after switching `_html_page`.

Reply:
My workaround for the issue:
```python
@property
def html_url(self):
    return self._html_url

@html_url.setter
def html_url(self, value):
    if self._html_url != value:
        self._html = None
        self.__class__.video_urls.fget.cache_clear()
        self.__class__.last_updated.fget.cache_clear()
        self.__class__.title.fget.cache_clear()
        self._html_url = value
```
It's ugly and would require syncing with every new cached property, but as a temporary solution works fine.

Reply:
I tested it and it's working fine your fix. I think that to work with several tabs it will be inevitable to update the cache